### PR TITLE
🏗  amp-timeago:1.0 - Use timeago.js library

### PR DESCRIPTION
**Pull request by caroqliu**
This PR:
- Migrates `amp-timeago 1.0` from using the `timeago` package in `third_party/` to the runtime dependency package
- Upgrades the version of timeago.js being used by amp-timeago:1.0 from 3.0.0 to 4.0.2
- Removes `third_party/timeagojs` as it's no longer needed given the above.
- Updates both versions of `amp-timeago` and corresponding docs to use `locale` values of the format `xx` or `xx-XX` (case-insensitive)  in accordance with [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt). Note: Prior formats (`xxXX`) are still supported but removed from docs to move away from non-standard formatting, and timeago.js library formats (`xx_XX`) were not added to the larger set of supported formats. 

This PR is a follow up to #32809. Related to #29246.

### CODE REVIEW

**Review by rsimha — APPROVED** [review]
Very nice! So much cleaner, with no real difference to binary size.

image

File: extensions/amp-timeago/1.0/component.js [hunk 1]
```diff
@@ -60,6 +60,11 @@ export function Timeago({
       return undefined;
     }
     const observer = new win.IntersectionObserver((entries) => {
+      let {lang} = node.ownerDocument.documentElement;
+      if (lang === 'unknown') {
```
Comment (dvoytenko) [inline extensions/amp-timeago/1.0/component.js:64]: Never seen "unknown", but I do see an empty string. There's also `navigator.language` that could be helpful for defaults.

**Review by dvoytenko — APPROVED** [review]
LGMT. But not seeing timeago package.json changes. Was that done already before?

File: extensions/amp-timeago/1.0/component.js (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-timeago/1.0/component.js:64]: I got `unknown` from this reference: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/lang, should I remove this check?

re: package.json changes, this was done in #32809 for 0.1 separately.

Comment (caroqliu) [reply extensions/amp-timeago/1.0/component.js:64]: Going to merge this PR for now, let's follow up separately if we decide to remove `"unknown"`

**Pull request merged**